### ISSUE TITLE
Use DI to provide Validator

### ISSUE THREAD

**Issue by jsobell** [aurelia/validatejs#47]
According to Rob the current implementation expects an instance of Validate:
```
constructor() {  
  this.validator = new Validator(this.model)
    .ensure('firstName')
      .required();
}
```
This is reported as being because the instance returned from the DI is a singleton, and a unique instance is required for each usage.
Can we have the plugin initialize process add the Validator to the DI as transient, so that the standard injection feature still applies?
Given that this is the only core Aurelia feature that requires 'new'ing, it's bound to be asked about repeatedly in the support areas.
The added advantage is that this also means the documentation is still correct!

**Comment by apawsey** [comment]
Alternatively, the singleton validator needs to start collecting all the rules for all objects it comes across.  I don't know if there's any significant downside to that, but if this was marked as transient, it would also surely be a deviation from the aurelia norm, as pretty much everything else is a singleton (event aggregator, router, httpclient, etc).

**Comment by EisenbergEffect** [comment]
I think there is an issue with the design preventing it from being used with DI actually. That fact that the Validator requires the model in its constructor causes this problem. I would personally still much rather enable the Validator to build up a set of rules and then "Apply" them to an object. This would allow DI to be used, with a transient registration. It would also allow the creation of rulesets which could be re-used across multiple objects.

**Comment by michaelmalonenz** [comment]
That sounds a lot like the:

```javascript
constructor(validtion) {
  this.validation = validation.on(this);
}
```

pattern :stuck_out_tongue:

**Comment by EisenbergEffect** [comment]
I'm going to close this issue because there a more comprehensive discussion elsehwere now which will resolve the final api for this.
